src/WeakRefStrings.jl

```diff
@@ -191,6 +193,15 @@ const MISSING_OFFSET = typemax(UInt64)-1
 Base.size(a::StringArray) = size(a.offsets)
 Base.IndexStyle(::Type{<:StringVector}) = IndexLinear()
 
+function DataAPI.refarray(a::StringArray{T}) where {T}
+    S = Union{WeakRefString{UInt8}, typeintersect(T, Missing)}
+    convert(StringArray{S}, a)
+end
+
+function DataAPI.refvalue(a::StringArray{T}, s) where {T}
```

Comment:
Wouldn't this work? Better be explicit I'd say.
```suggestion
function DataAPI.refvalue(a::StringArray{T}, s::AbstractString) where {T}
```

Reply:
That may be problematic as `missing` is also allowed in a `StringArray`. I've added `Union{AbstractString, Missing}` as signature.

Reply:
OK. Should we allow any `AbstractString`, though? AFAICT that's only supposed to be used with `WeakRefString{UInt8}`, right?

Reply:
What do you think of my suggestion?

Reply:
You're right, I've restricted the signature to `Union{WeakRefString{UInt8}, Missing}`, which encompasses all possible eltypes of `refarray`.